FutureNorthants/northants-design-system
# Add web chat bot form

See this issue for details: https://github.com/FutureNorthants/northants-website/issues/1030

The web chat will only be accessible from the contact page. This means that it needs to be a slice so it can be added as page content in the CMS. 

- [WebChat slice component](https://deploy-preview-496--northants-design-system.netlify.app/?path=/story/library-slices-webchat--example-web-chat)
- [Contact page preview](https://deploy-preview-496--northants-design-system.netlify.app/?path=/story/page-examples-contact-page--example-contact-page)

The frontend will need to implement code to check whether there are any agents available and what channels can be contacted. This is done when the user presses the 'Open WebChat' button. It is currently simulated by using a set timeout to show the loading spinner, but will need to be updated to contact the real API and get the required data. 

## Testing
- Checkout this branch and run `npm install` then `npm run dev` 
- View the Slices -> WebChat -> Example Web Chat story
- Update the action prop to the testing endpoint (ask in teams for it) then test completing the form.
- The name, subject and enquiry type should be required fields. It should not submit without these values.
- When you submit it should open a new tab and take you to the Storm web chat with the fields you completed in the url. The enquiry type should use the 'queue' query parameter.
